hyperspy/_signals/hologram_image.py

```diff
@@ -0,0 +1,329 @@
+                sb_position_temp = sb_position.data
+        else:
+            sb_position_temp = sb_position.deepcopy()
+        #
+
+        # Parsing sideband size
+        if sb_size is None:  # Default value is 1/2 distance between sideband and central band
+            if reference is None:
+                sb_size = self.find_sideband_size(sb_position)
+            else:
+                sb_size = reference.find_sideband_size(sb_position)
+        else:
+            if not isinstance(sb_size, BaseSignal):
+                if isinstance(sb_size, np.ndarray) and sb_size.size > 1:  # transpose if np.array of multiple instances
+                    sb_size = BaseSignal(sb_size).T
+                else:
+                    sb_size = BaseSignal(sb_size)
+
+        if sb_size.axes_manager.navigation_size != self.axes_manager.navigation_size:
+            if sb_size.axes_manager.navigation_size:
+                warnings.warn('Sideband size dimensions do not match neither reference nor hologram dimensions.'
+                              'The reconstruction will be performed with the first value.')
+                sb_size_temp = np.float64(sb_size.inav[0].data)
+            else:  # sb_position navdim=0, therefore map function should not iterate it:
+                sb_size_temp = np.float64(sb_size.data)
+        else:
+            sb_size_temp = sb_size.deepcopy()
+        #
+
+        # Standard edge smoothness of sideband aperture 5% of sb_size
+        if sb_smoothness is None:
+            sb_smoothness = sb_size * 0.05
+        else:
+            if not isinstance(sb_smoothness, BaseSignal):
+                if isinstance(sb_smoothness, np.ndarray) and sb_smoothness.size > 1:
+                    sb_smoothness = BaseSignal(sb_smoothness).T
+                else:
+                    sb_smoothness = BaseSignal(sb_smoothness)
+
+        if sb_smoothness.axes_manager.navigation_size != self.axes_manager.navigation_size:
+            if sb_smoothness.axes_manager.navigation_size:
+                warnings.warn('Sideband smoothness dimensions do not match neither reference nor hologram dimensions.'
+                              'The reconstruction will be performed with the first value.')
+                sb_smoothness_temp = np.float64(sb_smoothness.inav[0].data)
+            else:  # sb_position navdim=0, therefore map function should not iterate it:
+                sb_smoothness_temp = np.float64(sb_smoothness.data)
+        else:
+            sb_smoothness_temp = sb_smoothness.deepcopy()
+
+        # Convert sideband size from 1/nm or mrad to pixels
+        if sb_unit == 'nm':
+            sb_size_temp = sb_size_temp / np.mean(self.f_sampling)
+            sb_smoothness = sb_smoothness / np.mean(self.f_sampling)
+        elif sb_unit == 'mrad':
+            try:
+                ht = self.metadata.Acquisition_instrument.TEM.beam_energy
+            except AttributeError:
+                ht = int(input('Enter beam energy in kV: '))
```

Comment:
I'm not sure what is wrong there with the `set_microsocpe_parameters` but here is what I've got either switching to qt4 backend or initialising my standard OS X backend:

> In[3]: %matplotlib qt4
In[4]: aa = hs.signals.EELSSpectrum(np.reshape(np.arange(16), (2, 2, 2, 2)))
In[5]: aa
Out[5]: <EELSSpectrum, title: , dimensions: (2, 2, 2|2)>
In[6]: aa.set_microscope_parameters(beam_energy=300)
Traceback (most recent call last):
  File "/Users/vadim/anaconda/lib/python3.5/site-packages/IPython/core/interactiveshell.py", line 2885, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-6-2477a9aea25a>", line 1, in <module>
    aa.set_microscope_parameters(beam_energy=300)
  File "/Users/vadim/Work_Recent/Prog/_Py/hyperspy/hyperspy/_signals/eels.py", line 833, in set_microscope_parameters
    self._are_microscope_parameters_missing()
  File "/Users/vadim/Work_Recent/Prog/_Py/hyperspy/hyperspy/_signals/eels.py", line 793, in _are_microscope_parameters_missing
    is_ok = messagesui.information(par_str)
  File "/Users/vadim/Work_Recent/Prog/_Py/hyperspy/hyperspy/gui/messages.py", line 75, in information
    message.edit_traits()
  File "/Users/vadim/anaconda/lib/python3.5/site-packages/traits/has_traits.py", line 1859, in edit_traits
    handler, id, scrollable, args )
  File "/Users/vadim/anaconda/lib/python3.5/site-packages/traitsui/view.py", line 433, in ui
    ui.ui( parent, kind )
  File "/Users/vadim/anaconda/lib/python3.5/site-packages/traitsui/ui.py", line 222, in ui
    self.rebuild( self, parent )
  File "/Users/vadim/anaconda/lib/python3.5/site-packages/traitsui/toolkit.py", line 226, in ui_modal
    raise NotImplementedError
NotImplementedError
In[7]: aa.set_signal_type('EDS_SEM')
In[8]: aa
Out[8]: <EDSSEMSpectrum, title: , dimensions: (2, 2, 2|2)>
In[9]: aa.set_microscope_parameters(beam_energy=300)
In[10]: aa.metadata.Acquisition_instrument.SEM.beam_energy
Out[10]: 300